### ISSUE TITLE
SteamAPI is not available

### ISSUE THREAD

**Issue by mcwhy123** [renpy/renpy#5993]
renpy Version 8.3.3
I'm using achievement.steam.dlc_installed() in my code. This code works correctly, but running the exe file after packaging yields an error AttributeError: 'NoneType' object has no attribute 'dlc_installed'

**Comment by renpytom** [comment]
Please post log.txt.

**Comment by mcwhy123** [comment]
my log.txt




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "renpy/renpy"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年12月8日(星期天) 中午11:17
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [renpy/renpy] SteamAPI is not available (Issue #5993)





 
Please post log.txt.

**Comment by renpytom** [comment]
It didn't seem to go through, please post it on github.com.

**Comment by mcwhy123** [comment]
log.txt content：




2024-12-08 02:37:56 UTC
Windows-10-10.0.19045-SP0
Ren'Py 8.4.0.24120701+nightly


QianSe
1.0
Built at 2024-12-08 02:29:12 UTC


Early init took 0.06s
Loading error handling took 0.05s
56 script files skipped.
Loading script took 0.38s
Loading save slot metadata took 0.19s
Loading persistent took 0.01s
Failed to initialize steam: Exception('steamapi.InitFlat returned 1 (No appID found.&nbsp; Either launch the game from Steam, or put the file steam_appid.txt containing the correct appID in your game folder.)')
Set script version to: (8, 4, 0)
Running init code took 0.28s
Loading analysis data took 0.04s
Analyze and compile ATL took 0.01s
Reloading save slot metadata took 0.08s
Index archives took 0.00s
Dump and make backups took 0.00s
Cleaning cache took 0.00s
Making clean stores took 0.00s
Initial gc took 0.10s
DPI scale factor: 1.250000
nvdrs: Loaded, about to disable thread optimizations.
nvdrs: Disabled thread optimizations.
Creating interface object took 0.23s
Cleaning stores took 0.00s
Init translation took 0.09s
Build styles took 0.00s
Load screen analysis took 0.01s
Analyze screens took 0.00s
Save screen analysis took 0.00s
Prepare screens took 0.06s
Save pyanalysis. took 0.00s
Save bytecode. took 0.00s
Running _start took 0.00s
Interface start took 0.18s


Initializing gl2 renderer:
primary display bounds: (0, 0, 1920, 1080)
swap interval: 1 frames
Fullscreen mode.
Vendor: "b'NVIDIA Corporation'"
Renderer: b'GeForce GTX 1650/PCIe/SSE2'
Version: b'4.6.0 NVIDIA 462.75'
Display Info: None
Screen sizes: virtual=(1920, 1080) physical=(1920, 1080) drawable=(1920, 1080)
Maximum texture size: 4096x4096


Full traceback:
&nbsp; File "renpy/common/_layout/screen_main_menu.rpym", line 28, in script
&nbsp; &nbsp; python hide:
&nbsp; File "E:\创作归档\忘雪社\茜色\内测版\QianSe\QianSe-1.0-dists\QianSe-1.0-pc\renpy\ast.py", line 824, in execute
&nbsp; &nbsp; renpy.python.py_exec_bytecode(self.code.bytecode, self.hide, store=self.store)
&nbsp; File "E:\创作归档\忘雪社\茜色\内测版\QianSe\QianSe-1.0-dists\QianSe-1.0-pc\renpy\python.py", line 1189, in py_exec_bytecode
&nbsp; &nbsp; exec(bytecode, globals, locals)
&nbsp; File "renpy/common/_layout/screen_main_menu.rpym", line 28, in <module&gt;
&nbsp; &nbsp; python hide:
&nbsp; File "renpy/common/_layout/screen_main_menu.rpym", line 2011, in _execute_python_hide
&nbsp; File "E:\创作归档\忘雪社\茜色\内测版\QianSe\QianSe-1.0-dists\QianSe-1.0-pc\renpy\ui.py", line 301, in interact
&nbsp; &nbsp; rv = renpy.game.interface.interact(roll_forward=roll_forward, **kwargs)
&nbsp; File "E:\创作归档\忘雪社\茜色\内测版\QianSe\QianSe-1.0-dists\QianSe-1.0-pc\renpy\display\core.py", line 2201, in interact
&nbsp; &nbsp; repeat, rv = self.interact_core(preloads=preloads, trans_pause=trans_pause, pause=pause, pause_start=pause_start, pause_modal=pause_modal, **kwargs) # type: ignore
&nbsp; File "E:\创作归档\忘雪社\茜色\内测版\QianSe\QianSe-1.0-dists\QianSe-1.0-pc\renpy\display\core.py", line 2731, in interact_core
&nbsp; &nbsp; root_widget.visit_all(lambda d : d.per_interact())
&nbsp; File "E:\创作归档\忘雪社\茜色\内测版\QianSe\QianSe-1.0-dists\QianSe-1.0-pc\renpy\display\displayable.py", line 434, in visit_all
&nbsp; &nbsp; d.visit_all(callback, seen)
&nbsp; File "E:\创作归档\忘雪社\茜色\内测版\QianSe\QianSe-1.0-dists\QianSe-1.0-pc\renpy\display\displayable.py", line 434, in visit_all
&nbsp; &nbsp; d.visit_all(callback, seen)
&nbsp; File "E:\创作归档\忘雪社\茜色\内测版\QianSe\QianSe-1.0-dists\QianSe-1.0-pc\renpy\display\displayable.py", line 434, in visit_all
&nbsp; &nbsp; d.visit_all(callback, seen)
&nbsp; File "E:\创作归档\忘雪社\茜色\内测版\QianSe\QianSe-1.0-dists\QianSe-1.0-pc\renpy\display\screen.py", line 480, in visit_all
&nbsp; &nbsp; callback(self)
&nbsp; File "E:\创作归档\忘雪社\茜色\内测版\QianSe\QianSe-1.0-dists\QianSe-1.0-pc\renpy\display\core.py", line 2731, in <lambda&gt;
&nbsp; &nbsp; root_widget.visit_all(lambda d : d.per_interact())
&nbsp; File "E:\创作归档\忘雪社\茜色\内测版\QianSe\QianSe-1.0-dists\QianSe-1.0-pc\renpy\display\screen.py", line 491, in per_interact
&nbsp; &nbsp; self.update()
&nbsp; File "E:\创作归档\忘雪社\茜色\内测版\QianSe\QianSe-1.0-dists\QianSe-1.0-pc\renpy\display\screen.py", line 697, in update
&nbsp; &nbsp; self.screen.function(**self.scope)
&nbsp; File "game/screens.rpy", line 621, in execute
&nbsp; File "game/screens.rpy", line 621, in execute
&nbsp; File "game/screens.rpy", line 624, in execute
&nbsp; File "game/screens.rpy", line 2634, in <module&gt;
AttributeError: 'NoneType' object has no attribute 'dlc_installed'


While running game code:
&nbsp; File "game/screens.rpy", line 621, in execute
&nbsp; File "game/screens.rpy", line 621, in execute
&nbsp; File "game/screens.rpy", line 624, in execute
&nbsp; File "game/screens.rpy", line 2634, in <module&gt;
AttributeError: 'NoneType' object has no attribute 'dlc_installed'





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "renpy/renpy"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年12月8日(星期天) 中午12:11
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [renpy/renpy] SteamAPI is not available (Issue #5993)





 
It didn't seem to go through, please post it on github.com.

**Comment by renpytom** [comment]
The issue is that you don't have an appid set. You want to add a line like: 

```renpy
define config.steam_appid = 12345
```

Where 12345 is the appid steam assigned your game.

**Comment by mcwhy123** [comment]
I've set the config.steam_appid in options.rpy and achievement.steam.dlc_installed() works fine, but after building the distro, I run the QianSe.exe and I get the above problem (log.txt)

**Comment by renpytom** [comment]
The built distro needs to be launched through steam for it to work. (I believe you'll need to actually upload a test version to Steam and download it.)

**Comment by mcwhy123** [comment]
Thanks, I uploaded it and launched the game with steam successfully




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "renpy/renpy"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年12月8日(星期天) 中午1:38
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [renpy/renpy] SteamAPI is not available (Issue #5993)





 
The built distro needs to be launched through steam for it to work. (I believe you'll need to actually upload a test version to Steam and download it.)

**Comment by renpytom** [comment]
I'd strongly suggest conditioning uses of achivement.steam on it being not None. ie.:

```
python:
    if achievement.steam is not None and achievement.steam.dlc_installed(112358):
        ...
```

**Comment by mcwhy123** [comment]
Great advice, thank you




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "renpy/renpy"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年12月8日(星期天) 下午2:36
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [renpy/renpy] SteamAPI is not available (Issue #5993)





 
I'd strongly suggest conditioning uses of achivement.steam on it being not None. ie.:
 python:     if achievement.steam is not None and achievement.steam.dlc_installed(112358):         ...
